httpcore5/src/main/java/org/apache/hc/core5/http/impl/io/DefaultBHttpClientConnectionFactory.java

```diff
@@ -55,25 +56,44 @@
     private final CharCodingConfig charCodingConfig;
     private final ContentLengthStrategy incomingContentStrategy;
     private final ContentLengthStrategy outgoingContentStrategy;
+    private final ResponseOutOfOrderStrategy responseOutOfOrderStrategy;
     private final HttpMessageWriterFactory<ClassicHttpRequest> requestWriterFactory;
     private final HttpMessageParserFactory<ClassicHttpResponse> responseParserFactory;
 
-    public DefaultBHttpClientConnectionFactory(
```

Comment:
You cannot change this public interface imho.

Reply:
No but you can ADD a default method (Java 8); would that work here?

Reply:
Sorry for the confusion, the github diff is misleading. This private constructor is completely new, this method takes the additional ResponseOutOfOrderStrategy field. I opted to make it private so we can add future fields to the builder and we aren't forced to grow beyond five constructors.

Reply:
You are right, I see it now. The previous constructor remains `public`.